### ISSUE TITLE
Updating populated data causes multiple afterUpdates to trigger

### ISSUE THREAD

**Issue by xjpro** [balderdashy/sails#2163]
When a model contains an associated model, a socket update using fully populated data causes multiple afterUpdates lifecycles functions to trigger. A short demonstration:

Two models:

Person.js
```
module.exports = {
	attributes: {
		pet: {
			model: 'Pet'
		}
	},
	afterUpdate: function(person, next) {
		console.log('updating', new Date());
		next();
	}
};
```
Pet.js
```
module.exports = {
};
```

A simple socket PUT:
```
io.socket.put('/person/1', { pet: { id: 2 } });
```

This will cause afterUpdate for the Person model to trigger 4 times. However, this 

```
io.socket.put('/person/1', { pet: 2 });
```

will cause a single update, as expected. Essentially, submitting populated data in a PUT will cause multiple lifecycles to trigger. I would expect Person's afterUpdate to trigger once and Pet's afterUpdate to trigger once. 

This is for Sails 0.10.4.

**Comment by sgress454** [comment]
@xjpro thanks for the report, will check this out.  Keep in mind that nested updates (sending a whole update to the `PUT` blueprint) will probably be deprecated in the next Waterline release--they tend to cause more issues than they solve.

**Comment by sailsbot** [comment]
Thanks for posting, @xjpro.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
